### ISSUE TITLE
Use groups & sequences to validate order

### ISSUE THREAD

**Issue by alexsegura** [coopcycle/coopcycle-web#507]
In [`OrderValidator`](https://github.com/coopcycle/coopcycle-web/blob/dcfd7dbf60f8c2b88cc8c7fa196ef7398e8d5fac/src/AppBundle/Validator/Constraints/OrderValidator.php), we are currently using `return` statements to stop on the first error, with some logic. Looks like this can be achieved in a cleaner way using [Group Sequence Providers](https://symfony.com/doc/3.4/validation/sequence_provider.html). 

So basically, each constraint could be in its own class, and we could use groups.
Also, we can use [`ExecutionContextInterface::getObject`](https://api.symfony.com/3.4/Symfony/Component/Validator/Context/ExecutionContextInterface.html#method_getObject) to get the parent entity.
